### ISSUE TITLE
Registry configuration is invalid: Scoped registry at position 1 is invalid

### ISSUE THREAD

**Issue by CoderBeanLiang** [googleads/googleads-mobile-unity#1345]
Hello, I'm new to google ads for Unity, and was stuck in some problems for about 2 days.
I followed the Guide Page: [Doc](https://developers.google.com/admob/unity/start)

After imported GoogleMobileAds-v5.1.0.unitypackage, I was asked to add registries to Packages/manifest.json, which was:
------ name: Game Package Registry by Google,
-------url: **https://unityregistry-pa.googleapis.com**, scopes: [com.google]

I followed the instructions, added it then got some **errors**:
------------------------------------------------------------------------------------
Listing installed UPM packages failed (Registry configuration is invalid:
  Scoped registry at position 1 is invalid:
    Unable to connect "https://unityregistry-pa.googleapis.com" (InvalidParameter))
Registry configuration is invalid:
  Scoped registry at position 1 is invalid:
    Unable to connect "https://unityregistry-pa.googleapis.com"
Failed to resolve packages: Registry configuration is invalid:
  Scoped registry at position 1 is invalid:
    Unable to connect "https://unityregistry-pa.googleapis.com". No packages loaded.

A re-import of the project may be required to fix the issue or a manual modification of F:/UnityProjects/XXX/Packages/manifest.json file.
-----------------------------------------------------------------------------

Here was the configurations in my manifest.json:
-----------------------------------------------------------------------------
  "scopedRegistries": [
    {
      "name": "Game Package Registry by Google",
      "url": "https://unityregistry-pa.googleapis.com",
      "scopes": [
        "com.google"
      ]
    }
  ]
-----------------------------------------------------------------------------

I'm in China so I've tested with vpn proxy but no use. I directly visited it on browser via vpn but got 404error.
I can remove this scopedRegistries but the Package Manager Resolver always pops up to ask me to add this Google Package Registry.
I really donnot know what is it and how I can use it. Maybe my question is fool, but can anyone help me?
Thank you very much!

  * Unity version: 2019.3.6f1 or 2019.3.15f1
  * Google Mobile Ads Unity plugin version: v5.1.0
  * Platform: Win10

**Comment by stowy** [comment]
Hello @CoderBeanLiang - I am not aware of this game package registry by Google. I don't believe it is related to this plugin.

Are you adding additional packages to your project in another way?

Perhaps you can screenshot the prompt that suggested the adding of the registry?

**Comment by CoderBeanLiang** [comment]
@stowy Thank you for reply! And This is what pop up after imported plugin.
![PackageRegistry](https://user-images.githubusercontent.com/16380161/84458236-1e808600-ac97-11ea-8947-01224e96b83f.PNG)

If I click [Add Selected Registries], it will add some text in my Packages/manifest.json:
![jsoon](https://user-images.githubusercontent.com/16380161/84458889-a0bd7a00-ac98-11ea-87c4-b3733694917f.PNG)
Then the issue occured, some of the log just like this:
![isssus](https://user-images.githubusercontent.com/16380161/84459493-27268b80-ac9a-11ea-93f1-9604345d6e94.PNG)

If I click [Cancel] then it would pop up usually. What I can only do is [Disable Registry Addition].

I don't know what this registry is and all the info I can search in the network is https://developer.android.com/games/develop/build-in-unity#option_2_manually_edit_manifestjson
In this android game official page it mentioned the registry.

Additional: I created one new project and tested this plugin with v5.0.0release, this problem didn't appear.

**Comment by chkuang-g** [comment]
@CoderBeanLiang 

TL;DR; This is from [External Dependency Manager for Unity (EDM4U)](https://github.com/googlesamples/unity-jar-resolver). Even though it logged as an error, this should not affect your use of Google Ads plugin.

This is a new feature in [External Dependency Manager for Unity (EDM4U)](https://github.com/googlesamples/unity-jar-resolver) which comes with Google Ads package.  You can find all the plugin files under `Assets/ExternalDependencyManager` folder.

What it does is adding a registry hosted by Google to your Unity project so that you can manage some Google distributed packages through [Unity Package Manager (UPM)](https://docs.unity3d.com/Manual/upm-ui.html). And EDM4U is one of it.  After you click on `Add selected Registies`, EDM4U will attempt to migrate itself to be installed through UPM.

According to the error message, Unity was not able to connect to the registry while attempting to migrate the package.  I suspect that either the registry was down by that time, or your machine was offline. There is no harm if it failed to begin the migration process.  What it means is that EDM4U will remains in `Assets` folder instead of `Packages` folder.

What we can do probably is to detect the situation and perhaps lower the severity of the log so that it would look less scary.

Could you confirm if everything still function properly even Unity logged this error?

**Comment by CoderBeanLiang** [comment]
@chkuang-g Really thanks for your detailed reply.
Caz your help now I am clearly aware of this registry, and with selecting [Disable Registry Addition] my ads works perfectly.
I think waht you say are totally right, even though I can open your url by browser and see a json text with many google packages, but there must be something wrong with my Unity Network settings, I will recheck it.
Thanks again for your help, I'll close this issue.
